librz/bin/bobj.c

```diff
@@ -1017,3 +995,119 @@ RZ_API ut64 rz_bin_object_v2p(RzBinObject *obj, ut64 vaddr) {
 	}
 	return m->paddr + delta;
 }
+
+RZ_IPI RZ_OWN RzBinStrDb *rz_bin_string_database_new(RzList *list, ut64 load_address, bool decode_base64) {
+	RzBinStrDb *db = RZ_NEW0(RzBinStrDb);
+	if (!db) {
+		RZ_LOG_ERROR("rz_bin: Cannot allocate RzBinStrDb\n");
+		rz_list_free(list);
+		return NULL;
+	}
+
+	db->list = list ? list : rz_list_newf((RzListFree)rz_bin_string_free);
+	db->phys = ht_up_new0();
+	db->virt = ht_up_new0();
+	if (!db->list || !db->phys || !db->virt) {
+		RZ_LOG_ERROR("rz_bin: Cannot allocate RzBinStrDb internal data structure.\n");
+		goto fail;
+	}
+
+	RzListIter *it;
+	RzBinString *bstr;
+	rz_list_foreach (list, it, bstr) {
+		bstr->paddr += load_address;
+		if (decode_base64) {
+			char *decoded = bstr->string;
+			do {
+				// ensure to decode base64 strings encoded multiple times.
+				char *tmp = (char *)rz_base64_decode_dyn(decoded, -1);
+				if (!tmp || !rz_str_is_printable(tmp)) {
+					free(tmp);
+					break;
+				}
+				free(decoded);
+				decoded = tmp;
+			} while (1);
+			if (decoded != bstr->string) {
+				free(bstr->string);
+				bstr->string = decoded;
+				bstr->type = RZ_STRING_ENC_BASE64;
+			}
+		}
+
```

Comment:
I do wonder whether this should be done here or somewhere else. I mean, there could be several types of encoding and shall they all be done here? Maybe it should be the responsibility of the caller to convert the strings in the right encoding.

Reply:
this was already part of the old code, i just integrated it in the loop since it was a small optimization

Reply:
I know there was a separate function for this however as you are defining new apis I wonder whether it's really the right place for this.